src/views/Users/ViewUser/index.jsx

```diff
@@ -117,7 +121,9 @@ function ViewUser({ isNewUser, ...props }) {
 
   useEffect(() => {
     Promise.all([
-      isNewUser ? null : fetchUser(existingUsername),
+      userExists(existingUsername).then(exists =>
+        exists ? fetchUser(existingUsername) : null
+      ),
```

Comment:
Instead of using `userExists`, any chance we can use `withUser`? I think we would be able to rewrite to something like:

```jsx
Promise.all([
  user ? fetchUser(existingUsername) : Promise.resolve(),
  // ...
]);
```

Reply:
Unless I'm misunderstanding, I don't think that will work here, because the username in question isn't the logged in user, it's the user we're creating or editing.

Reply:
I think fetching a user that doesn't exists should show an error to the user as a form of action feedback.

Also, the issue I was having was on the view that lists all users, not a single user. I don't seem to get this error anymore for some reason.

Reply:
Okay. I still think we need to do something here though - there was an error that did show up on ViewUser for a user that only has scheduled permissions.

Reply:
> there was an error that did show up on ViewUser for a user that only has scheduled permissions.

Would it make sense to have the backend not throw and error but instead return an empty set when performing the request that is currently throwing the error?

Reply:
Thinking from the backend perspective I don't think this makes sense - trying to fetch a user that doesn't exist seems like a legitimate 404 to me. I'll see if I can come up with another way to address this.

Reply:
I agree with you. Maybe we can just catch the error. We could probably use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled when FF supports it but for the time being one approach I think we do is take the asynchronous call outside the `Promise.all` so that it doesn't stop the other calls for proceeding.

Reply:
> Maybe we can just catch the error

I was looking into this angle, and our interceptor already catches and throws a new Error: https://github.com/mozilla-frontend-infra/balrog-ui/blob/master/src/App.jsx#L32. We could catch that, of course, but without the actual axios response we have to check the error message itself to know whether it was a 404 or something else, which is pretty nasty. This makes me wonder if the interceptor needs to change...